### PR TITLE
Add first 2 screens from login

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

**Review by eli1stark — CHANGES_REQUESTED** [review]
Good work, but needs some improvements

File: lib/auth/view/components/spot_outlined_button.dart [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    required this.onTap,
+  }) : super(key: key);
+
+  final String text;
+  final void Function() onTap;
+
+  @override
+  Widget build(BuildContext context) {
+    final authScreen = useProvider(authScreenStpod);
+    final notifier = useProvider(authenticationNotipod.notifier);
```
Comment (eli1stark) [inline lib/auth/view/components/spot_outlined_button.dart:21]: try to make view abstract from state management

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/auth/view/components/spot_outlined_button.dart [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+import '../../provider/auth_screen_stpod.dart';
+import '../../provider/authentication_notipod.dart';
+
+class SpotOutlinedButton extends HookWidget {
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/components/spot_outlined_button.dart:8]: done

File: lib/auth/view/components/filled_button.dart [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import 'package:flutter/material.dart';
+
+class SpotFilledButton extends StatelessWidget {
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/components/filled_button.dart:3]: done

File: lib/auth/view/components/text_fields/email_text_field.dart [hunk 4]
```diff
@@ -15,13 +10,28 @@ class EmailTextField extends HookWidget {
...
-      style: baseFieldStyle,
-      decoration: emailFieldDecoration(intl),
+      decoration: const InputDecoration(
+        border: UnderlineInputBorder(),
+        hintText: 'Email address',
+      ),
     );
   }
+
+  String? _validate(String? value) {
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/components/text_fields/email_text_field.dart:25]: refactored

File: lib/auth/view/components/text_fields/password_text_field.dart [hunk 5]
```diff
@@ -17,20 +13,36 @@ class PasswordTextField extends HookWidget {
...
+          ),
+          onPressed: () {
+            isObscure.value = !isObscure.value;
+          },
+        ),
       ),
     );
   }
+
+  String? _validate(String? value) {
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/components/text_fields/password_text_field.dart:39]: refactored

File: lib/main.dart [hunk 6]
```diff
@@ -57,7 +56,7 @@ class MyApp extends HookWidget {
       localizationsDelegates: AppLocalizations.localizationsDelegates,
       supportedLocales: AppLocalizations.supportedLocales,
       debugShowCheckedModeBanner: false,
-      theme: appTheme,
+      // theme: appTheme,
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/main.dart:59]: removed

File: lib/auth/view/welcome.dart [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+            onTap: () => navigatorPush(
+                context,
+                const SignInSignUp(
+                  isSignIn: false,
+                )),
+          ),
+          const SpaceH15(),
+          SpotOutlinedButton(
+            text: 'I have an account',
+            onTap: () => navigatorPush(
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/welcome.dart:47]: refactored

File: lib/auth/view/welcome.dart [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+            textAlign: TextAlign.start,
+            style: TextStyle(
+              fontWeight: FontWeight.bold,
+              fontSize: 46,
+            ),
+          ),
+          Expanded(child: Container()),
+          SpotFilledButton(
+            text: 'Create account',
+            onTap: () => navigatorPush(
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/welcome.dart:38]: refactored

File: lib/auth/view/welcome.dart [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+                context,
+                const SignInSignUp(
+                  isSignIn: false,
+                )),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/welcome.dart:42]: refactored

File: lib/auth/view/welcome.dart [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+          ),
+          const SpaceH15(),
+          SpotOutlinedButton(
+            text: 'I have an account',
+            onTap: () => navigatorPush(
+                context,
+                const SignInSignUp(
+                  isSignIn: true,
+                )),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/welcome.dart:51]: refactored

File: lib/auth/view/welcome.dart [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+class Welcome extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(24),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          Expanded(child: Container()),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/welcome.dart:18]: refactored

File: lib/auth/view/sign_in_sign_up.dart [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+                      ),
+                      PasswordTextField(
+                        controller: credentials.passwordController,
+                      ),
+                      Expanded(child: Container()),
+                      SpotFilledButton(
+                        text: 'Continue',
+                        onTap: () {
+                          notifier.authenticate(
+                              isSignIn ? AuthScreen.signIn : AuthScreen.signUp);
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/sign_in_sign_up.dart:84]: refactored

File: lib/auth/view/sign_in_sign_up.dart [hunk 13]
```diff
@@ -0,0 +1,98 @@
...
+                      const SpaceH30(),
+                      const Text(
+                        'Enter password',
+                        textAlign: TextAlign.start,
+                        style: TextStyle(),
+                      ),
+                      PasswordTextField(
+                        controller: credentials.passwordController,
+                      ),
+                      Expanded(child: Container()),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/sign_in_sign_up.dart:79]: refactored

File: lib/auth/view/welcome.dart [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+          ),
+          const Text(
+            'to the app',
+            textAlign: TextAlign.start,
+            style: TextStyle(
+              fontWeight: FontWeight.bold,
+              fontSize: 46,
+            ),
+          ),
+          Expanded(child: Container()),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/welcome.dart:35]: refactored

File: lib/auth/view/welcome.dart [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+class Welcome extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Padding(
+      padding: const EdgeInsets.all(24),
+      child: Column(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: <Widget>[
+          Expanded(child: Container()),
+          const Text(
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/welcome.dart:19]: refactored

File: lib/auth/view/sign_in_sign_up.dart [hunk 16]
```diff
@@ -0,0 +1,98 @@
...
+                        style: TextStyle(),
+                      ),
+                      EmailTextField(
+                        controller: credentials.emailController,
+                      ),
+                      const SpaceH30(),
+                      const Text(
+                        'Enter password',
+                        textAlign: TextAlign.start,
+                        style: TextStyle(),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/sign_in_sign_up.dart:74]: refactored

File: lib/auth/view/sign_in_sign_up.dart [hunk 17]
```diff
@@ -0,0 +1,98 @@
...
+                        style: const TextStyle(
+                          fontWeight: FontWeight.bold,
+                          fontSize: 32,
+                        ),
+                      ),
+                      const SpaceH20(),
+                      const Text(
+                        'Enter your email',
+                        textAlign: TextAlign.start,
+                        style: TextStyle(),
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/sign_in_sign_up.dart:65]: refactored

File: lib/auth/view/sign_in_sign_up.dart [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+                      ),
+                      EmailTextField(
+                        controller: credentials.emailController,
+                      ),
+                      const SpaceH30(),
+                      const Text(
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/sign_in_sign_up.dart:71]: left as it is

File: lib/auth/view/sign_in_sign_up.dart [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+                      Text(
+                        isSignIn ? 'Log In' : 'Create an account',
+                        textAlign: TextAlign.start,
+                        style: const TextStyle(
+                          fontWeight: FontWeight.bold,
+                          fontSize: 32,
+                        ),
+                      ),
+                      const SpaceH20(),
+                      const Text(
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/auth/view/sign_in_sign_up.dart:62]: left as it is

File: lib/auth/view/components/spot_outlined_button.dart (hunk 1 above)
Comment (VovaMedvediev) [reply lib/auth/view/components/spot_outlined_button.dart:21]: done

**Pull request merged**